infogami/plugins/wikitemplates/code.py

```diff
@@ -44,6 +48,13 @@ def __getitem__(self, key):
 
         return value
 
+    # TODO: Should __iter__() and keys() return the same keys??
+    def __iter__(self):
+        return self.templates.__iter__()
+
+    def __len__(self):
+        return len(self.templates)
+
```

Comment:
Can you give a quick summary of these are now necessary? Is it because we switched to `Mapping`?

Reply:
http://python3porting.com/problems.html#replacing-userdict